node/src/components/storage.rs

```diff
@@ -1851,14 +1855,15 @@ impl Storage {
                         return Ok(None);
                     }
                 };
-
             result.push(parent_block_header.clone());
             if parent_block_header.is_switch_block() {
-                break;
+                depth += 1;
+                if depth >= era_depth || parent_block_header.is_genesis() {
```

Comment:
On mainnet, genesis is not a switch block, so I think we would end up returning `Ok(None)`.
Not sure if we're handling that case correctly yet?